core/chain-configs/src/genesis_config.rs

```diff
@@ -175,11 +177,29 @@ impl From<&GenesisConfig> for EpochConfig {
 )]
 pub struct GenesisRecords(pub Vec<StateRecord>);
 
+#[derive(Debug, Copy, Clone)]
+pub enum GenesisRecordsFileType {
+    FullGenesis,
+    RecordsArray,
+}
+
+#[derive(Debug, Clone)]
+pub struct GenesisRecordsFile {
+    pub path: PathBuf,
+    pub file_type: GenesisRecordsFileType,
+}
```

Comment:
Am I correct that `path` only makes sense when the type is `RecordsArray`? If that's the case, let's make illegal states unrepresentable: 

```suggestion
#[derive(Debug, Copy, Clone)]
pub enum GenesisRecordsFile {
    FullGenesis,
    RecordsArray { path: PathBuf },
}
```

Reply:
Ah, reading more into this, I see that's not the case: `Path` might point either to the genesis file itself, or to a separaet file with records.